### DISCUSSION TITLE
searchCatalogItems overly particular

### DISCUSSION THREAD

**Discussion by MojoJake** [amzn/selling-partner-api-models#3839] [General]
Has anyone else found that searchCatalogItems (https://developer-docs.amazon.com/sp-api/docs/catalog-items-api-v2022-04-01-reference#searchcatalogitems) is extremely exacting in returning results? Or perhaps I'm using it wrong? I do note that it changed from MWS "ListMatchingProducts" when the parameter was a "query" to now its a list of "keywords" which could account for the difference.

1) If I search for "Howard the amazing Mr. Hughes Dietrich" I get no results. Removing "Dietrich", which is the author, causes the book to show up. Why would a search of books that includes the author's name not work? I find this in many examples where including the author name to try and narrow the results causes it to just return nothing.

2) When searching for the book "What is History?" and typing "what is istory" which has a clear type-o, you get 0 results. Obviously you can sort that out by cleaning up your search and adding the missing "H" in "History", but that is VERY exacting considering any modern search should be able to make those connections.

Both of the above examples returned relevant results in MWS.

**Comment by callumb123** [comment]
I have found the same, the old ListCatalogItem has always been superior to SearchCatalogItems.

I noticed a similar thing today when adding '30cm' to the end of the query. Without it it returns all of the results I would expect (including the expected product which has 30cm in the title). When I add the '30cm' then all of a sudden it returns an entirely different set of results and now misses the expected product despite the query being an even closer match as it also now matches on 30cm.

Weird.

**Comment by chapmanjw** [comment]
@MojoJake Can you provide request ids and timestamps for example requests where the same query on ListCatalogItems and SearchCatalogItems produces different results? They are ultimately both powered by the same search index as the retail websites. With examples, we can deep dive into what might be triggering a difference.

**Reply by chapmanjw** [reply]
With the first search example (`Howard the amazing Mr. Hughes Dietrich`), I am seeing the same one result whether I pass it as one keywords entry with spaces or as a list of words: `&keywords=Howard%20the%20amazing%20Mr.%20Hughes%20Dietrich` or `&keywords=Howard,the,amazing,Mr.,Hughes,Dietrich`.

```
{
  "numberOfResults": 1,
  "refinements": {
    "brands": [
      {
        "numberOfResults": 1,
        "brandName": "Fawcett"
      }
    ],
    "classifications": [
      {
        "numberOfResults": 7,
        "displayName": "Books",
        "classificationId": "1000",
        "classificationPaths": [
          [
            {
              "displayName": "Books",
              "classificationId": "1000"
            }
          ]
        ]
      }
    ]
  },
  "items": [
    {
      "asin": "B0006WM7O4",
      "summaries": [
        {
          "marketplaceId": "ATVPDKIKX0DER",
          "adultProduct": false,
          "autographed": false,
          "browseClassification": {
            "displayName": "Biographies & Memoirs",
            "classificationId": "2"
          },
          "contributors": [
            {
              "role": {
                "displayName": "Author",
                "value": "author"
              },
              "value": "Dietrich, Noah"
            }
          ],
          "itemClassification": "BASE_PRODUCT",
          "itemName": "Howard, the amazing Mr. Hughes (A Fawcett gold medal book)",
          "manufacturer": "Fawcett Publications",
          "memorabilia": false,
          "tradeInEligible": false,
          "websiteDisplayGroup": "book_display_on_website",
          "websiteDisplayGroupName": "Book"
        }
      ]
    }
  ]
}
```

**Reply by chapmanjw** [reply]
For the `what is istory` example, what you're seeing is expected behavior. We are relaying the search query exactly as-is to the index to adhere to what the client is asking for. We could, potentially, expand on this in the future to enable suggested query fixes (similar to the notion of "did you mean _What is History_" you see on some UI-based search pages). I will forward the idea to our product team to review as a potential future feature to add to the roadmap.
